File: Android_Easylayer_Java/app/src/main/java/com/example/easylayer_110_gs300/Sat/Rede.java
package com.example.easylayer_110_gs300.Sat;

import android.app.AlertDialog;
import android.content.Context;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import com.example.easylayer_110_gs300.R;
import com.example.satfuncoes.SatFunctions;

import java.util.Random;

public class Rede extends AppCompatActivity {
    private Button button_configurar;
    private EditText txtCodAtivacao, txtIp, txtMascara, txtGateway, txtDns, txtDns2, txtProxyIp, txtPorta, txtUser, txtPassword;
    private Spinner spinnerTipoRede, spinnerOptDns, spinnerOptProxy;
    private AlertDialog alerta;
    private static final String[] tipoRede = new String[]{"Estático", "DHCP"};
    private static final String[] optDns = new String[]{"Não", "Sim"};
    private static final String[] optProxy = new String[]{"Não usa proxy", "Proxy com configuração", "Proxy transparente"};
    private String appPackage = "com.gertec.exemplosgertec8";
    private String caminhoXML = "data/data/" + appPackage + "/configRede.xml";
    Random gerador = new Random();
    Context context;
    SatFunctions satFunctions;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_rede);

    }
}

File: Android_Nativo_Kotlin/app/src/main/java/com/example/gs300_2023/sat/Rede.kt
package com.example.gs300_2023.sat

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Spinner
import br.estudos.satfuncoes.SatFunctions
import com.example.gs300_2023.R
import java.util.Random



class Rede : Activity() {
    private lateinit var button_configurar: Button
    private lateinit var txtCodAtivacao: EditText
    private lateinit var txtIp: EditText
    private lateinit var txtMascara: EditText
    private lateinit var txtGateway: EditText
    private lateinit var txtDns: EditText
    private lateinit var txtDns2: EditText
    private lateinit var txtProxyIp: EditText
    private lateinit var txtPorta: EditText
    private lateinit var txtUser: EditText
    private lateinit var txtPassword: EditText
    private lateinit var spinnerTipoRede: Spinner
    private lateinit var spinnerOptDns: Spinner
    private lateinit var spinnerOptProxy: Spinner
    private lateinit var alerta: android.app.AlertDialog

    companion object {
        val tipoRede = arrayOf("Estático", "DHCP")
        val optDns = arrayOf("Não", "Sim")
        val optProxy = arrayOf("Não usa proxy", "Proxy com configuração", "Proxy transparente")
        const val appPackage = "com.gertec.exemplosgertec8"
        val caminhoXML = "data/data/$appPackage/configRede.xml"
        val gerador = Random()
    }

    private lateinit var context: Context
    private lateinit var satFunctions: SatFunctions

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.rede)
        context = this@Rede
        satFunctions = SatFunctions(this)

    }
}
